crates/api/src/op_store.rs

```diff
@@ -91,6 +91,22 @@ pub trait OpStore: 'static + Send + Sync + std::fmt::Debug {
         op_ids: Vec<OpId>,
     ) -> BoxFuture<'_, K2Result<Vec<MetaOp>>>;
 
+    /// Retrieve a size-bounded list of op ids that have been stored since `start`.
+    ///
+    /// The `limit_bytes` applies to the size of the op data, not the size of the op ids.
+    ///
+    /// # Returns
+    ///
+    /// As many op ids as can be returned within the `limit_bytes` limit.
+    ///
+    /// If the limit is applied, then the timestamp of the last op id returned is returned.
+    /// Otherwise, the timestamp for when this operation started is returned.
```

Comment:
What is this operation start time used for?

Reply:
This is to do with the incremental get of ops by stored_at. The start time is the bookmark that the other peer provided to us to fetch ops from that point in time. I can improve naming here maybe to make that clearer?

Reply:
I thought so, but it doesn't immediately make sense, because one peer could be requesting ops from a week ago with a limit, then get half of them with the last op's timestamp, and get the other half with the timestamp of when the operation started which is a couple of seconds ago maybe.
I see that it is probably an operation over _all_ ops that exist since the start_time until now, so the above case would only happen if there are no more ops after the second batch of them. Then it makes sense and could be aided by a short comment.

Reply:
I don't follow. The timestamp bookmark is maintained per peer by each peer. So it's not going to move unpredictably. It's not a perfect mechanism because people will drop their bookmarks while offline and then we have to rely on the DHT module to get us back in sync.

Can we maybe discuss your example so I can figure out what it would be useful to add to this documentation?

Reply:
To me it would be useful to have a comment such as
"The returned timestamp is kept in memory as a bookmark of up until when an agent has requested ops from a peer. If only a portion of all ops from `start` until now have been exchanged, the timestamp of the last exchanged op is returned. Otherwise all ops have already been exchanged and the time of the request serves as the bookmark."

The bit that I didn't understand at first was that the retrieval of ops is across _all_ ops until now.

Reply:
Alright, I see what you're getting at. I should be explaining that this is operating on the "store_at" timestamp so that this is a "paging" operation. I won't explain how gossip uses the function here, but I do want to make it clear how this function is meant to operate

Reply:
Updated here -> https://github.com/holochain/kitsune2/pull/72/commits/e61dfbc3645efb8ebe2fcde43bf289bc8f56af2e